gatewayApp/gateway/node_connection_client.py

```diff
@@ -0,0 +1,278 @@
+import socket
+import time
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+from cryptography.hazmat.primitives.hashes import SHA256
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+import os
+import sys
+
+import node_connection
+
+sys.path.insert(0, '../../Message')
+from Message import *
+
+# TODO: replace with actual address
+HOST = '127.0.0.1'
+PORT = 65432
+
+ATTEMPTS = 3
+REFRACTORY_PERIOD = 1 #[s]
+
+# Helper functions for sending/receiving data
+def send_data(socket: socket, data: bytes) -> None:
+    """Sends data to Node
+
+    Args:
+        sock (socket): Socket for communication
+        data (bytes): Data to send.
+    """
+    try:
+        # Send the length of the data (4 bytes, big-endian)
+        socket.send(len(data).to_bytes(4, 'big'))
+        # Send the actual data
+        socket.sendall(data)
+    except Exception as e:
+        print(e)
+        raise
+    
+def receive_data(socket: socket) -> bytes:
+    """Retrieves message from Node.
+
+    Args:
+        sock (socket): Socket for communication.
+
+    Raises:
+        ConnectionError: Connection with Node broken.
+
+    Returns:
+        bytes: Encoded message.
+    """    
+    data_length: int = int.from_bytes(socket.recv(4), 'big')
```

Comment:
Do we have to specify this directly? I think it will obviously be int.
I mean:
```suggestion
    data_length = int.from_bytes(socket.recv(4), 'big')
```

Same later